test/regress_dns.c

```diff
@@ -848,6 +848,103 @@ dns_retry_disable_when_inactive_test(void *arg)
 	dns_retry_test_impl(arg, EVDNS_BASE_DISABLE_WHEN_INACTIVE);
 }
 
+static void
+dns_probe_settings_test_impl(void *arg, int flags)
+{
+	struct basic_test_data *data = arg;
+	struct event_base *base = data->base;
+	struct evdns_server_port *port = NULL;
+	struct evdns_base *dns = NULL;
+	int drop_count = 1;
+	ev_uint16_t portnum = 0;
+	char buf[64];
+
+	struct generic_dns_callback_result r1, r2;
+	struct timeval tvalBefore, tvalAfter;
+	unsigned long delta;
+
+	port = regress_get_udp_dnsserver(base, &portnum, NULL,
+	    fail_server_cb, &drop_count);
+	tt_assert(port);
+	evutil_snprintf(buf, sizeof(buf), "127.0.0.1:%d", (int)portnum);
+
+	dns = evdns_base_new(base, flags);
+	tt_assert(!evdns_base_nameserver_ip_add(dns, buf));
+	tt_assert(! evdns_base_set_option(dns, "timeout", "0.2"));
+	tt_assert(! evdns_base_set_option(dns, "max-timeouts:", "1"));
+	tt_assert(! evdns_base_set_option(dns, "attempts:", "1"));
+	tt_assert(! evdns_base_set_option(dns, "initial-probe-timeout", "10"));
+	// it will also set initial-probe-timeout to 1s (10s > 1s)
+	tt_assert(! evdns_base_set_option(dns, "max-probe-timeout:", "1"));
+
+	evdns_base_resolve_ipv4(dns, "host.example.com", 0,
+	    generic_dns_callback, &r1);
+	n_replies_left = 2;
+	exit_base = base;
+	gettimeofday (&tvalBefore, NULL);
+	// this will wait until the probe request done
+	// should be around 1s instead of 10s
+	event_base_dispatch(base);
+	gettimeofday (&tvalAfter, NULL);
+	tt_int_op(r1.result, ==, DNS_ERR_TIMEOUT);
+	delta = ((tvalAfter.tv_sec - tvalBefore.tv_sec)*1000000L
+					+tvalAfter.tv_usec) - tvalBefore.tv_usec;
+	tt_assert(delta / 1000000L == 1);
+
+	// should be ok now
+	evdns_base_resolve_ipv4(dns, "host.example.com", 0,
+	    generic_dns_callback, &r1);
+	n_replies_left = 1;
+	event_base_dispatch(base);
+
+	tt_int_op(r1.result, ==, DNS_ERR_NONE);
+	tt_int_op(r1.type, ==, DNS_IPv4_A);
+	tt_int_op(r1.count, ==, 1);
+	tt_int_op(((ev_uint32_t*)r1.addrs)[0], ==, htonl(0x10204080));
+
+	// dns server down again
+	drop_count = 3;
+	tt_assert(! evdns_base_set_option(dns, "max-probe-timeout:", "3600"));
+	tt_assert(! evdns_base_set_option(dns, "initial-probe-timeout", "0.2"));
+	tt_assert(! evdns_base_set_option(dns, "probe-backoff-factor:", "10"));
+	evdns_base_resolve_ipv4(dns, "host.example.com", 0, generic_dns_callback, &r1);
+	evdns_base_resolve_ipv4(dns, "host.example.com", 0, generic_dns_callback, &r2);
+	// probe timeout should be around 2s now 
+	n_replies_left = 3;
+
+	gettimeofday (&tvalBefore, NULL);
+	// wait dns server up
+	event_base_dispatch(base);
+	tt_int_op(r1.result, ==, DNS_ERR_TIMEOUT);
+	tt_int_op(r2.result, ==, DNS_ERR_TIMEOUT);
+	gettimeofday (&tvalAfter, NULL);
```

Comment:
```suggestion
	gettimeofday(&tvalAfter, NULL);
```

Maybe I missed some bits, but you got the idea